### PR TITLE
Add simple qint8 quantization for the Phi-3 Model

### PR DESCRIPTION
**Pull request by dev-sungman**
Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/TBD-Labs-AI/latched/blob/repo-setting/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

# Description
This PR introduces classes for the optimizers. To check this, I added some lines to the example code.
Currently, quint8 quantization for the Phi-3.5 is working, however, it only works well to the weight. 

# Changes

- Change the basic concept: `Model` -> `ModelWrapper`. Huggingface Model includes the `model` itself and maybe other libraries also similar. So, if we use the `Model`, then we need to call `Model.model.model...`. To avoid this, I renamed the `Model`->`ModelWrapper`. And `AutoModelWrapper` will resolve the recursive `.model` call.
- Add some skeleton class for further implementation `SmoothQuant`
- Add `ptq`, `qat` folder inside the `model_optimizers`

# Usage example

Added at a example code

# Benefits
- 

# Testing
- 

# Next steps
- Add Model Exporter to finish the whole pipeline

### CODE REVIEW

File: src/latched/model_optimizers/auto.py [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
-    import torch.nn as nn
+import torch.nn as nn
 
 
 class AutoOptimizer(BaseModelOptimizer):
     """AutoOptimizer to use the optimization techniques with a easiest way."""
 
     @classmethod
-    def run(model: nn.Module) -> None:
+    def run(cls, model: nn.Module) -> nn.Module:
```
Comment (psycoder-sup) [inline src/latched/model_optimizers/auto.py:13]: 이거를 classmethod로 wrap 하신 이유가 뭐에용? 잘몰라서... ㅎㅎ

**Review by psycoder-sup — APPROVED** [review]
궁금한거 하나 질문 넣었습니다. ㅎㅎ

File: src/latched/model_optimizers/auto.py (hunk 1 above)
Comment (sungmanch) [reply src/latched/model_optimizers/auto.py:13]: 유저 사용성을 고려했는데요. 아래처럼 두줄로 표현하는 것 보다는 classmethod로 한번에 처리하는게 더 사용하기 편하고, 굳이 선언해서 사용할 이유 없겠다고 생각해서 이렇게 구현했습니당 ~

```python3
auto_optimizer = AutoOptimizer()
optimized_model = auto_optimizer.run(model)
```

Comment (psycoder-sup) [reply src/latched/model_optimizers/auto.py:13]: 화긴화긴

**Pull request merged**